website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -4913,6 +5227,28 @@ If you would like to set a default coder, use the method described in the
 previous section, *Setting the default coder for a type*.
 {{< /paragraph >}}
 
+{{< paragraph class="language-rust" >}}
+The Beam SDK for Rust does not support annotating data types with a default coder.
+Instead, you can define a custom type which implements `ElemType` trait.
+The `ElemType` trait requires you to define a standard coder.
+
+```rust
+struct YourType {}
+
+impl ElemType for YourType {
+    type StandardCoder = YourTypeStandardCoder;
+    // ...
+}
+
+struct YourTypeStandardCoder {}
+
+impl CoderI for YourTypeStandardCoder {
+    type E = YourType;
+    // ...
+}
+```
+{{< /paragraph >}}
+
```

Comment:
If we move this paragraph into 7.2.2, then maybe we could have a paragraph here about how we provide a derive macro to automatically derive the ElemType for any type built out of other ElemTypes... WDYT?

Reply:
Derive macro for ElemType is a cool feature! I will add some description about it in section 7.2.3.